backend/corpora/lambdas/api/v1/curation/collections/collection_uuid/datasets/actions.py

```diff
@@ -0,0 +1,39 @@
+import re
+from flask import g, make_response
+
+from backend.corpora.api_server.db import dbconnect
+from backend.corpora.common.corpora_orm import CollectionVisibility
+from backend.corpora.common.entities import Dataset
+from backend.corpora.common.utils.http_exceptions import InvalidParametersHTTPException, ConflictException
+from backend.corpora.common.utils.regex import DATASET_ID_REGEX, CURATOR_TAG_NAME_REGEX, EXTENSION_REGEX
+from backend.corpora.lambdas.api.v1.authorization import owner_or_allowed
+from backend.corpora.lambdas.api.v1.common import get_dataset_else_error, get_collection_else_forbidden
+
+REGEX = f"^({DATASET_ID_REGEX}|{CURATOR_TAG_NAME_REGEX})\\.{EXTENSION_REGEX}$"
+
+
+def validate_curator_tag(curator_tag):
+    matched = re.match(REGEX, curator_tag)
+    return True if matched and matched.groupdict().get("tag") else False
+
+
+@dbconnect
+def patch(token_info: dict, collection_uuid: str, body: dict, curator_tag: str = None, dataset_uuid: str = None):
+    db_session = g.db_session
+    dataset = get_dataset_else_error(db_session, dataset_uuid, collection_uuid, curator_tag)
+    get_collection_else_forbidden(
+        db_session, collection_uuid, visibility=CollectionVisibility.PRIVATE, owner=owner_or_allowed(token_info)
+    )
+    tag = body["curator_tag"]
```

Comment:
Related to the issue of `tag.h5ad` vs `tag`: what if the curator submits `arbitrary_tag.h5ad` as the tag? In order to overwrite/update that Dataset via the local upload, they would need to upload to `<s3_key_prefix>/arbitrary_tag.h5ad.h5ad`.

Reply:
I think we should, in accordance with general "path" rules, prohibit the `.` character from being contained within the tag.

Reply:
> Related to the issue of `tag.h5ad` vs `tag`: what if the curator submits `arbitrary_tag.h5ad` as the tag? In order to overwrite/update that Dataset via the local upload, they would need to upload to `<s3_key_prefix>/arbitrary_tag.h5ad.h5ad`.

Actually, they would upload to `<s3_key_prefix>/arbitrary_tag.h5ad` to update the dataset with tag `arbitrary_tag.h5ad`. If they wanted to upload to `<s3_key_prefix>/arbitrary_tag.h5ad.h5ad` they could and it would be valid. Not recommended, but I don't see why we need to include the logic to prevent this.